### PR TITLE
ci(gcb): tag builds as ci or pr

### PR DESCRIPTION
**Pull request by devjgm**
These tags will make querying our builds easier. This also renames PR builds to include `-pr` in the name.

Fixes: #6182

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6215)

### CODE REVIEW

File: ci/cloudbuild/triggers/quickstart-bazel-ci.yaml [hunk 1]
```diff
@@ -7,4 +7,6 @@ github:
 name: quickstart-bazel-ci
 substitutions:
   _BUILD_NAME: quickstart-bazel
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/quickstart-bazel-ci.yaml:10]: nit: this seems unrelated, I am Okay merging it now.  If it was not intentional, please fix. If it was intentional, consider separating to another PR (but again, I am Okay either way).

**Review by devbww — APPROVED** [review]

File: ci/cloudbuild/triggers/tsan-pr.yaml [hunk 2]
```diff
@@ -5,7 +5,9 @@ github:
   pullRequest:
     branch: ^master$
     commentControl: COMMENTS_ENABLED_FOR_EXTERNAL_CONTRIBUTORS_ONLY
-name: tsan
+name: tsan-pr
 substitutions:
   _BUILD_NAME: tsan
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/tsan-pr.yaml:11]: I think this is not going to work, we need clang-9 for the TSAN build, right?

File: ci/cloudbuild/triggers/tsan-ci.yaml [hunk 3]
```diff
@@ -7,4 +7,6 @@ github:
 name: tsan-ci
 substitutions:
   _BUILD_NAME: tsan
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/tsan-ci.yaml:10]: ditto regarding TSAN and clang-9.

File: ci/cloudbuild/triggers/quickstart-cmake-pr.yaml [hunk 4]
```diff
@@ -5,7 +5,9 @@ github:
   pullRequest:
     branch: ^master$
     commentControl: COMMENTS_ENABLED_FOR_EXTERNAL_CONTRIBUTORS_ONLY
-name: quickstart-cmake
+name: quickstart-cmake-pr
 substitutions:
   _BUILD_NAME: quickstart-cmake
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/quickstart-cmake-pr.yaml:11]: intentional?

File: ci/cloudbuild/triggers/quickstart-cmake-ci.yaml [hunk 5]
```diff
@@ -7,4 +7,6 @@ github:
 name: quickstart-cmake-ci
 substitutions:
   _BUILD_NAME: quickstart-cmake
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/quickstart-cmake-ci.yaml:10]: intentional?

File: ci/cloudbuild/triggers/quickstart-bazel-pr.yaml [hunk 6]
```diff
@@ -5,7 +5,9 @@ github:
   pullRequest:
     branch: ^master$
     commentControl: COMMENTS_ENABLED_FOR_EXTERNAL_CONTRIBUTORS_ONLY
-name: quickstart-bazel
+name: quickstart-bazel-pr
 substitutions:
   _BUILD_NAME: quickstart-bazel
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/quickstart-bazel-pr.yaml:11]: intentional?

File: ci/cloudbuild/triggers/gcc-54-pr.yaml [hunk 7]
```diff
@@ -5,7 +5,9 @@ github:
   pullRequest:
     branch: ^master$
     commentControl: COMMENTS_ENABLED_FOR_EXTERNAL_CONTRIBUTORS_ONLY
-name: gcc-54
+name: gcc-54-pr
 substitutions:
   _BUILD_NAME: gcc-54
   _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/gcc-54-pr.yaml:11]: I did not notice this before, but does fedora have GCC 5.4?

File: ci/cloudbuild/triggers/gcc-54-ci.yaml [hunk 8]
```diff
@@ -8,3 +8,5 @@ name: gcc-54-ci
 substitutions:
   _BUILD_NAME: gcc-54
   _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/gcc-54-ci.yaml:10]: Ditto regarding fedora and gcc-5.4

File: ci/cloudbuild/triggers/cmake-super-pr.yaml [hunk 9]
```diff
@@ -5,7 +5,9 @@ github:
   pullRequest:
     branch: ^master$
     commentControl: COMMENTS_ENABLED_FOR_EXTERNAL_CONTRIBUTORS_ONLY
-name: cmake-super
+name: cmake-super-pr
 substitutions:
   _BUILD_NAME: cmake-super
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/cmake-super-pr.yaml:11]: intentional?

File: ci/cloudbuild/triggers/cmake-super-ci.yaml [hunk 10]
```diff
@@ -7,4 +7,6 @@ github:
 name: cmake-super-ci
 substitutions:
   _BUILD_NAME: cmake-super
-  _DISTRO: ubuntu-bionic
+  _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/cmake-super-ci.yaml:10]: intentional?

File: ci/cloudbuild/triggers/clang-38-pr.yaml [hunk 11]
```diff
@@ -5,7 +5,9 @@ github:
   pullRequest:
     branch: ^master$
     commentControl: COMMENTS_ENABLED_FOR_EXTERNAL_CONTRIBUTORS_ONLY
-name: clang-38
+name: clang-38-pr
 substitutions:
   _BUILD_NAME: clang-38
   _DISTRO: fedora
```
Comment (coryan) [inline ci/cloudbuild/triggers/clang-38-pr.yaml:11]: Does fedora have Clang-3.8?

**Review by coryan — CHANGES_REQUESTED** [review]
I think there are some accidental changes mixed in?  Also some stuff around fedora vs. ubuntu that is probably Okay but you should double check.

File: ci/cloudbuild/triggers/quickstart-bazel-ci.yaml (hunk 1 above)
Comment (devjgm) [reply ci/cloudbuild/triggers/quickstart-bazel-ci.yaml]: These `_DISTRO` changes were all accidental. Good catch! All fixed/reverted back to `ubuntu-bionic` in the places that changed.

File: ci/cloudbuild/triggers/gcc-54-pr.yaml (hunk 7 above)
Comment (devjgm) [reply ci/cloudbuild/triggers/gcc-54-pr.yaml:11]: Nope. That's a bug that I introduced in https://github.com/googleapis/google-cloud-cpp/pull/6135. I'll fix that in a followup PR.

File: ci/cloudbuild/triggers/gcc-54-ci.yaml (hunk 8 above)
Comment (devjgm) [reply ci/cloudbuild/triggers/gcc-54-ci.yaml:10]: Nope. That's a bug that I introduced in #6135. I'll fix that in a followup PR.

File: ci/cloudbuild/triggers/clang-38-pr.yaml (hunk 11 above)
Comment (devjgm) [reply ci/cloudbuild/triggers/clang-38-pr.yaml:11]: Nope. That's a bug that I introduced in #6135. I'll fix that in a followup PR.

**Review by coryan — APPROVED** [review]
LGTM, thanks.
